pythonic-query.py

```diff
@@ -0,0 +1,172 @@
+from datetime import datetime
+import time
+import subprocess
+import json
+import requests
+
+studyid_counter = 0
+formatted_datetime = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+
+def clean_text(text):
+    if isinstance(text, str):
+        text = text.replace(" :", "").replace(": ", "").replace(":", "")  # remove misplaced colons
+        text = text.replace('"', "'")  # replace double quotes with single quotes
+        text = text.lstrip(", ")  # remove leading commas and spaces
+        if text.startswith(">"):  # ensure '>' values are properly quoted
+            text = f'"{text}"'
+    return text
+
+
+# remove duplicate keys in properties
+def remove_duplicates(properties):
+    unique_properties = {}
+    for prop_name, prop_values in properties.items():
+        unique_properties[prop_name] = list(set(prop_values))
+    return unique_properties
+
+
+# fetch the list of experiment accessions dynamically from GXA API
+def get_experiment_ids():
+    bash_command = 'curl -sS https://www.ebi.ac.uk/gxa/json/experiments | jq -r ".experiments[].experimentAccession" | sort -u'
+    experiment_accessions = subprocess.check_output(bash_command, shell=True, text=True)
+    study_ids = experiment_accessions.strip().split("\n")
+
+    return study_ids
+
+
+def fetch_and_parse_data(study_id):
+    study_url = f"https://www.ebi.ac.uk/gxa/json/experiments/{study_id}"
+
+    max_retries = 5
+    retry_delay = 5
+    attempt = 0
+
+    while attempt < max_retries:
+        try:
+            response = requests.get(study_url)
+            response.raise_for_status()
+
+            # check if the response is empty or not valid JSON
+            if not response.text.strip():
+                attempt += 1
+                time.sleep(retry_delay)
+                continue
+
+            # try to load the json data from the response
+            try:
+                json_data = response.json()
+            except json.JSONDecodeError:
+                attempt += 1
+                time.sleep(retry_delay)
+                continue
+
+            # if json is valid, process the data
+            if isinstance(json_data, dict) and "experiment" in json_data:
+                break
+
+        except requests.RequestException:
+            attempt += 1
+            time.sleep(retry_delay)
+
+    experiment_type = json_data.get("experiment", {}).get("type", "N/A")
+    organism = json_data.get("experiment", {}).get("species", "N/A")
+
+    global studyid_counter
+    studyid_counter += 1
+
+    # print study header
+    print(f"  - accession_count: {studyid_counter}")
+    print(f"    accession: {study_id}")
+    print(f"    experiment_type: {experiment_type}")
+    print(f"    organism: {organism}")
+    print("    assay_groups:")
```

Comment:
one easy way to avoid this hardcoded spaces (which can be error prone) is to create dict and use `yaml` library to create yaml file

i.e 
```
with open("example.yaml", "w") as file:
    yaml.dump(data, file, default_flow_style=False, sort_keys=False)
```

Reply:
comment addressed here: [b8cace9](https://github.com/ebi-gene-expression-group/atlas-experiment-fetcher/pull/1/commits/b8cace9c9f3ed69b5d837b60b771acd4044c5ce5)